### PR TITLE
gnuplot 5.0.4 recipe no longer works, libgd PNG

### PR DESCRIPTION
**Pull request by peterjc**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [X] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Including the license_file meta data ought to be useful as it is a non-standard license.

The secondary purpose is I want to see if the current recipe actually still works or not (given problems with gnuplot 4.6.0 on #4655 and in updating to gnuplot 5.0.5 in #4730).

### CODE REVIEW

**Comment by bgruening** [comment]
I think this is good as it is. But please keep in mind that such a commit does not trigger a rebuild - which is fine in this case.

**Comment by peterjc** [comment]
Sadly as I suspected, the recipe itself is now stale in some way and fails to build with PNG support. Linux and Mac both say:

```
** Configuration summary for gnuplot 5.0.4:
gnuplot will be compiled with the following terminals:
...
  libgd-based png, jpeg, and gif terminals: no (see config.log) 
```

**Comment by peterjc** [comment]
@croth1 do we need a new BioConda ``libgd`` with PNG support? Should it be named differently to prevent this problem happening again?

Or can we depend on https://anaconda.org/bioconda-legacy/libgd/files version libgd-2.1.1.bioconda-1 explicitly (under the legacy channel)?

**Comment by croth1** [comment]
> @croth1 do we need a new BioConda ``libgd`` with PNG support?

No, as I mentioned above, you have to depend on libgd >=2.2. This will pull in a libgd version that works with gnuplot.


> Or can we depend on https://anaconda.org/bioconda-legacy/libgd/files version libgd-2.1.1.bioconda-1 explicitly (under the legacy channel)?

No, please don't. bioconda's libgd version is in legacy for a reason. It has been completely replaced by conda-forge's version of libgd.

**Comment by peterjc** [comment]
It looks like I rewrote the history on #4730, but I was sure using `` libgd >=2.2`` made no difference. But I'll try that again here to be sure...

**Comment by peterjc** [comment]
OK, that fixed the Mac OS X build 👍 

But Linux is failing, of potential interest:

```
/opt/conda/conda-bld/recipe_1495485736413/work/gnuplot-5.0.4/missing: Unknown `--is-lightweight' option
Try `/opt/conda/conda-bld/recipe_1495485736413/work/gnuplot-5.0.4/missing --help' for more information
configure: WARNING: 'missing' script is too old or missing
```

```
checking for LIBCERF... configure: WARNING:
Package requirements (libcerf) were not met:
No package 'libcerf' found
Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.
Alternatively, you may set the environment variables LIBCERF_CFLAGS
and LIBCERF_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```

```
undefined reference to symbol 'libiconv_open'
```

File: recipes/gnuplot/meta.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ package:
   version: {{ version }}
 
 build:
-  number: 0
+  number: 1
```
Comment (croth1) [inline recipes/gnuplot/meta.yaml]: no need to change the build number here.

File: recipes/gnuplot/meta.yaml [hunk 2]
```diff
@@ -19,16 +19,17 @@ requirements:
     - gcc [not osx]
     - llvm [osx]
     - pkg-config [osx]
```
Comment (croth1) [inline recipes/gnuplot/meta.yaml]: You should use `pkg-config` also on linux

**Comment by peterjc** [comment]
I don't really mind which Conda channel ``gnuplot`` comes from, as long as it comes with PNG support.

Should the BioConda gnuplot be deprecated to encourage using the conda-forge version instead?

**Comment by bgruening** [comment]
Peter could you rebase this branch against master?

**Comment by peterjc** [comment]
Rebased as requested.

**Comment by peterjc** [comment]
It looks like the recipe's PNG test fails on a font issue - so big progress with the compilation issue - nice!

```
$ mkdir noarch
$ echo '{}' > noarch/repodata.json
$ bzip2 -k noarch/repodata.json
+ source /opt/conda/bin/activate /opt/conda/conda-bld/gnuplot_1497609128232/_t_env
+ /bin/bash -x -e /opt/conda/conda-bld/gnuplot_1497609128232/test_tmp/run_test.sh
+ gnuplot -e 'set terminal dumb; set style histogram; p '\''test-data.txt'\'''
                                                                               
                                                                               
    5 +-+------+-------+--------+--------+-------+--------+-------+------+-A   
      +        +       +        +        +       +        +       +        +   
  4.5 +-+                                          'test-data.txt'    A  +-+   
      |                                                                    |   
      |                                                                    |   
    4 +-+                                                 A              +-+   
      |                                                                    |   
  3.5 +-+                                                                +-+   
      |                                                                    |   
    3 +-+                                A                               +-+   
      |                                                                    |   
      |                                                                    |   
  2.5 +-+                                                                +-+   
      |                                                                    |   
    2 +-+              A                                                 +-+   
      |                                                                    |   
      |                                                                    |   
  1.5 +-+                                                                +-+   
      +        +       +        +        +       +        +       +        +   
    1 A-+------+-------+--------+--------+-------+--------+-------+------+-+   
      1       1.5      2       2.5       3      3.5       4      4.5       5   
                                                                               
+ gnuplot -e 'set terminal png'
fontconfig: Couldn't find font. when opening font "arial", using internal non-scalable font
INFO:conda_build.config:--dirty flag not specified.  Removing build folder after successful build/test.
```

Note the ``make check`` output is very verbose we might want to skip that (since it worked).

**Comment by bgruening** [comment]
Interesting, or is this the error? `./prepare: line 24: aclocal: command not found`

**Comment by cbrueffer** [comment]
gnuplot was removed in favour of the conda-forge version, I think this PR can be closed?

**Comment by peterjc** [comment]
OK, let's leave this to conda-forge (and hopefully get their package to build and test PNG support). Thanks

**Pull request closed without merging**